barretenberg/cpp/src/barretenberg/vm2/simulation/lib/written_slots_tree.hpp

```diff
@@ -0,0 +1,54 @@
+#pragma once
+
+#include "barretenberg/vm2/common/field.hpp"
+#include "barretenberg/vm2/simulation/lib/db_interfaces.hpp"
+#include "barretenberg/vm2/simulation/lib/indexed_memory_tree.hpp"
+
+namespace bb::avm2::simulation {
+
+struct WrittenPublicDataSlotLeafValue {
```

Comment:
This seems to comply with some kind of expectations (e.g. `is_updatable`, `get_key` not returning a `const&`...) could you add a comment/docstring?